src/libsync/vfs/cfapi/hydrationjob.cpp

```diff
@@ -184,11 +271,36 @@ void OCC::HydrationJob::onNewConnection()
     Q_ASSERT(!_transferDataSocket);
     Q_ASSERT(!_job);
 
-    qCInfo(lcHydration) << "Got new connection starting GETFileJob" << _requestId << _folderPath;
-    _transferDataSocket = _transferDataServer->nextPendingConnection();
-    _job = new GETFileJob(_account, _remotePath + _folderPath, _transferDataSocket, {}, {}, 0, this);
-    connect(_job, &GETFileJob::finishedSignal, this, &HydrationJob::onGetFinished);
-    _job->start();
+    if (isEncryptedFile()) {
+        // TODO: the following code is borrowed from PropagateDownloadEncrypted (should we factor it out and reuse? YES! Should we do it now? Probably not, as, this would imply modifying PropagateDownloadEncrypted, so we need a separate PR)
```

Comment:
I would have expected that to be done in `GETEncryptedFileJob` class to help with refactoring
does that make sense to you ?
why did you not do it ? I may be missing something
that would simplify HydrationJob and would remove some e2ee related code in VFS CfApi plugin

Reply:
@mgallien no, we need a separate class for that, because, `PropagateDownlodEncrypted` is also using the same code, and, we will end up duplicating it. So, we need a separate Job that we can then reuse in both `GETEncryptedFileJob` and `PropagateDownloadEncrypted`. Or, did you have another plan in mind? Could you provide a draft here of how you would refactor this?